'test_importable_module_skip[__main__-False]' is failing when run via 'python -m pytest'

I do realize I might be doing something uncommon but when tests are run via `python -m pytest`, the following test is failing:

```pytb
_____________________________________________ test_importable_module_skip[__main__-False] _____________________________________________

word = '__main__', expected = False

    @pytest.mark.parametrize(
        "word,expected",
        [
            ('os', True),
            ('os.name', False),
            ('__main__', False),
            ("don't", False),
        ]
    )
    def test_importable_module_skip(word, expected):
        f = filters.ImportableModuleFilter(None)
>       assert f._skip(word) is expected
E       AssertionError: assert True is False
E        +  where True = <bound method ImportableModuleFilter._skip of <sphinxcontrib.spelling.filters.ImportableModuleFilter object at 0x7f6cc66f4eb0>>('__main__')
E        +    where <bound method ImportableModuleFilter._skip of <sphinxcontrib.spelling.filters.ImportableModuleFilter object at 0x7f6cc66f4eb0>> = <sphinxcontrib.spelling.filters.ImportableModuleFilter object at 0x7f6cc66f4eb0>._skip

tests/test_filter.py:87: AssertionError
```

It works fine when starting `pytest` via the entry point script. Not sure if you consider this a bug but it feels a bit inconsistent that two roughly equivalent methods of starting pytest give different results.

(we are using `pythonX.Y -m pytest` as an easy way of forcing pytest to run via a specific Python version)

--- a/docs/source/history.rst
+++ b/docs/source/history.rst
@@ -6,6 +6,18 @@
 
    unmaintained
 
+7.3.2
+=====
+
+Bug Fixes
+---------
+
+- `#141 <https://github.com/sphinx-contrib/spelling/pull/141>`__ Treat
+  `__main__` as a special module name that cannot be imported. If the
+  test suite is invoked by running `python -m pytest` instead of
+  `pytest` then there will be no `__main__` and find_spec() will fail,
+  so this change makes the tests work in both modes.
+
 7.3.1
 =====
 

--- a/sphinxcontrib/spelling/filters.py
+++ b/sphinxcontrib/spelling/filters.py
@@ -180,6 +180,12 @@ class ImportableModuleFilter(Filter):
         super().__init__(tokenizer)
         self.found_modules = set(sys.builtin_module_names)
         self.sought_modules = self.found_modules.copy()
+        # By adding __main__ to the list of sought modules but not
+        # found modules we ensure that it is never recognized as a
+        # valid module, which is consistent with the behavior before
+        # version 7.3.1.  See
+        # https://github.com/sphinx-contrib/spelling/issues/141
+        self.sought_modules.add('__main__')
 
     def _skip(self, word):
         valid_module_name = all(n.isidentifier() for n in word.split('.'))

